### ISSUE TITLE
New-PowerBIWorkspace: Operation returned an invalid status code 'BadRequest'

### ISSUE THREAD

**Issue by wesleyathexagon** [microsoft/powerbi-powershell#400]
I'm trying to run the `New-PowerBIWorkspace` cmdlet using the credentials of a service principal. Here are the steps I follow to reproduce the issue:
``` powershell
# 1. Setup Credentials
$pbiApplicationId="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
$pbiApplicationSecret="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
$pbiApplicationSecret = ConvertTo-SecureString -String $pbiApplicationSecret -AsPlainText -Force
$psCred = New-Object System.Management.Automation.PSCredential -ArgumentList ($pbiApplicationId, $pbiApplicationSecret)
$pbiTenantId = "xxxxxxxxxxxxxxxxxxxxxxxxx"
$pbiEnvironment = "Public"
$pbiWorkspace = "mytestworkspace"

# 2. connect the service account
Connect-PowerBIServiceAccount -Environment $pbiEnvironment -ServicePrincipal -Credential $psCred -Tenant $pbiTenantId

# 3. Try to create the workspace
New-PowerBIWorkspace -Name $pbiWorkspace -Verbose
```

When I do that, I get this error:

``` powershell
PS /home/user> New-PowerBIWorkspace -Name $pbiWorkspace -Verbose
New-PowerBIWorkspace: Operation returned an invalid status code 'BadRequest'
VERBOSE: Request Uri: https://api.powerbi.com/v1.0/myorg/groups
VERBOSE: Status Code: BadRequest (400)
```

My service service principal seems to have the required permissions to run this. Besides, it's not an authorization error, it's just a generic 400 error. Is this possibly a problem with the powershell module?

**Comment by allankichowdary** [comment]
Any inputs here.

**Comment by wesleyathexagon** [comment]
I was able to resolve this issue. The workspace name I was trying to create already existed in my tenant but I did not have permissions to it. I found another user in my organization who had created a workspace with the same name I was trying to create. After the other person deleted their workspace, I was able to run the command without error.

It would have been helpful to get a 403 error or maybe an error that says, "workspace by that name already exists". But I'm aware that responsibility may not be on the powerbi powershell module.
